ui/src/shared/components/NewAnnotation.js

```diff
@@ -0,0 +1,211 @@
+import React, {Component, PropTypes} from 'react'
+import classnames from 'classnames'
+import {connect} from 'react-redux'
+import uuid from 'node-uuid'
+
+import OnClickOutside from 'shared/components/OnClickOutside'
+import AnnotationWindow from 'shared/components/AnnotationWindow'
+import * as schema from 'shared/schemas'
+import * as actions from 'shared/actions/annotations'
+
+class NewAnnotation extends Component {
+  state = {
+    isMouseOver: false,
+    gatherMode: 'startTime',
+  }
+
+  enforceGraphBounds = newTime => {
+    const {dygraph} = this.props
+    const xRangeStart = dygraph.xAxisRange()[0]
+
+    return newTime < xRangeStart ? `${xRangeStart}` : `${newTime}`
```

Comment:
Using Math.max would be easier to read.

Reply:
I realized I can destructure `[xStart]` from `dygraph.xAxisRange()` so much better. Was doing this elsewhere